lesson_13/dragla_marius/lesson13_inheritance.py

```diff
@@ -0,0 +1,86 @@
+class PerimeterMixin:
+
+    def perimeter(self):
+        return sum(self.sides)
+
+class Polygons(PerimeterMixin):
+
+    def __init__(self, *args):
+        self.sides = args
+
+    def __str__(self):
+        no_of_sides = len(self.sides)
+        return f'Number of sides: {no_of_sides}'
+
+    def display(self):
+        for side_idx, side_length in enumerate(self.sides, start=1):
+            print (f'Side {side_idx} with length: {side_length}')
+
+
+class Triangles(Polygons):
+    # s(s-a)(s-b)(s-c) ** 0.5` where s = `(a+b+c) / 2
+    def __init__(self, *args):
+        if len(args) > 3:
+            raise Exception(f'Shape is not a triangle. Number of sides: {len(args)}')
+        else:
+            super().__init__(*args)
+
+    def area(self):
+        s1, s2, s3 = self.sides
+        s_p = sum(self.sides) / 2
+        return f'Triangle area is {int((s_p*(s_p-s1)*(s_p-s2)*(s_p-s3)) ** 0.5)}'
+
+    def __len__(self):
+        return len(self.sides)
+
+
+class Square(Polygons):
+
+    def __init__(self, *args):
+        super().__init__(*args)
+
+    def area(self):
+        side, *_ = self.sides
+        return f'Square area is {side ** 2}'
+
+    def from_area(self):
```

Comment:
This method is supposed to return a new Square object. The logic of getting the sides is correct, but with the sides, we need to create a new object. In order to create the new object we need the class object (cls). 

What we can use from python to achive this?

Reply:
used cls